source/mainnet/smart-contracts/guides/unit-test-contract.rst

```diff
@@ -482,11 +482,193 @@ account:
         let amount1 = Amount::from_ccd(20);
         claim_eq!(host.get_transfers_to(receiver0), [amount0, amount1]);
 
+.. _writing_property_based_tests:
+
+Writing property-based tests
+============================
+
+The property-based testing technique allows for testing statements about your code that are expected to be true for any input parameters.
+The input to such tests is generated randomly.
```

Comment:
Could you emphasize here what randomly entails. That one has to write a generator that generates meaningful inputs. And that the efficacy of the testing is heavily dependent on this.

Reply:
I have a note at the end of the docs after the example. Should I move it here?

Reply:
I converted a note into a warning a moved it right after I write about `Arbitrary`